transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java

```diff
@@ -0,0 +1,137 @@
+     * necessary.
+     *
+     * @param apiValue data in raw form
+     * @return a String formatted to the specification in the mask pattern
+     */
+    protected String formatToDisplay(@NonNull final String apiValue) {
+        if (mField != null && mField.getMask() != null) {
+            // format
+            String pattern = mField.getMask().getPattern(apiValue);
+            if (!TextUtils.isEmpty(pattern)) {
+                return format(apiValue, pattern);
+            } else {
+                return apiValue;
+            }
+        }
+        return apiValue;
+    }
+
+    private String format(@NonNull final String apiValue, @NonNull final String pattern) {
+        if (apiValue.length() == 0 || pattern.length() == 0) {
+            return "";
+        }
+        StringBuilder formattedValue = new StringBuilder();
+        String extraTokens = "";
+        int patternIndex = 0;
+        int textIndex = 0;
+
+        while (true) {
+            char token = pattern.charAt(patternIndex);
+            char textChar = apiValue.charAt(textIndex);
+
+            switch (token) {
+                case NUMBER_TOKEN:
+                    if (Character.isDigit(textChar)) {
+                        if (extraTokens.length() > 0) {
+                            formattedValue.append(extraTokens);
+                            extraTokens = "";
+                        }
+                        formattedValue.append(textChar);
+                        patternIndex++;
+                    }
+                    textIndex++;
+                    break;
+                case TEXT_TOKEN:
+                    if (Character.isLetter(textChar)) {
+                        if (extraTokens.length() > 0) {
+                            formattedValue.append(extraTokens);
+                            extraTokens = "";
+                        }
+                        formattedValue.append(textChar);
+                        patternIndex++;
+                    }
+                    textIndex++;
+                    break;
+                case LETTER_OR_NUMBER_TOKEN:
+                    if (Character.isDigit(textChar) || Character.isLetter(textChar)) {
+                        if (extraTokens.length() > 0) {
+                            formattedValue.append(extraTokens);
+                            extraTokens = "";
+                        }
+                        formattedValue.append(textChar);
+                        patternIndex++;
+                    }
+                    textIndex++;
+                    break;
+                default:
+                    extraTokens += token;
+                    patternIndex++;
+            }
+            if (textIndex >= apiValue.length() || patternIndex >= pattern.length()) {
+                break;
+            }
+        }
+        return formattedValue.toString();
+    }
+
+    @VisibleForTesting
+    String formatForTesting(@NonNull final String apiValue, @NonNull final String pattern) {
```

Comment:
in this case I'd just make ethe format method package/default and remove this method